Dev 3.19

GigaZiv opened the pull request:
Выполнено домашнее задание согласно 19 спирта

С уважением Жуков Игорь

GigaZiv committed:
Dev 3.19

Выполнено домашнее задание согласно 19 спирта

С уважением Жуков Игорь

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/library/ui/fragments/LibraryFragment.kt:
```diff
@@ -0,0 +1,44 @@
+
+class LibraryFragment : Fragment() {
+
+    private lateinit var binding: FragmentLibraryBinding
```
При использовании `viewBinding` во фрагментах обязательно нужно обнулять `_binding` в `onDestroyView`, иначе будет утечка памяти. Почитать о том, как использовать эту фичу во фрагментах можно тут - https://developer.android.com/topic/libraries/view-binding#fragments. Почитать подробнее про утечку можно здесь - https://stackoverflow.com/questions/65295104/android-view-binding-clear-binding-in-fragment-lifecycle

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/search/ui/fragments/SearchFragment.kt:
```diff
@@ -0,0 +1,211 @@
+
+    private val viewModel by viewModel<SearchViewModel>()
+    private lateinit var binding: FragmentSearchBinding
+    private val tracks = mutableListOf<Track>()
```
Данные лучше хранить в стейте вьюмодели, не стоит дублировать список во фрагменте

ArturNurtdinov reviewed (commented):
1. Нижняя панель навигации не соответствует макетам в светлой и тёмной теме - нужно исправить высоту, цвет бэкграунда и цвет активного элемента:

<img width="853" height="247" alt="image" src="https://github.com/user-attachments/assets/52bee27a-64cb-474f-96a9-ad928f339a65" />
<img width="855" height="300" alt="image" src="https://github.com/user-attachments/assets/94215e17-0d62-405f-89eb-b3df27f26535" />

Макет:

<img width="954" height="293" alt="image" src="https://github.com/user-attachments/assets/b4b85230-a857-463d-b679-3d4fd4ca5240" />
<img width="954" height="281" alt="image" src="https://github.com/user-attachments/assets/835403ab-a1f2-499e-a30e-ba8a75175237" />

2. TabLayout и экраны избранных треков и медиатеки не соответствуют макетам - нужно поправить цвет TabLayout в тёмной теме и исправить плейсхолдеры избранных треков и плейлистов:

<img width="599" height="585" alt="image" src="https://github.com/user-attachments/assets/88d0f65a-57f1-406b-b1c3-bf30a04e8d3a" />
<img width="600" height="527" alt="image" src="https://github.com/user-attachments/assets/5f9fb21f-f389-4830-b61b-63dea71ee33c" />

Макет:

<img width="580" height="593" alt="image" src="https://github.com/user-attachments/assets/1acc318b-a88c-45c6-b1b6-f8faf131c8a7" />
<img width="576" height="647" alt="image" src="https://github.com/user-attachments/assets/14617a85-b9bb-4e59-8af4-5ccdc83ca9ed" />

GigaZiv committed:
Dev 3.19.1

Исправлены замечания согласно 19 спирта

С уважением Жуков Игорь

ArturNurtdinov commented:
Цвет TabLayout и цвет нижней панели навигации в тёмной теме не соответствуют макетам:

<img width="383" height="776" alt="image" src="https://github.com/user-attachments/assets/842f116f-5669-447f-a6d4-76f37550aa13" />

Макет:

<img width="367" height="759" alt="image" src="https://github.com/user-attachments/assets/d5e9427d-66a6-4771-8a76-8226dbc76446" />

GigaZiv committed:
Dev 3.19.2

Прошу прощения за глупые ошибки, очень спешу и времени крайне мало.

Исправлены замечания согласно 19 спирта

С Уважением Жуков Игорь

С уважением Жуков Игорь
